src/plans/create/steps/GeneralInformationStep.tsx

```diff
@@ -0,0 +1,203 @@
+                  onSelect={(_, value) => field.onChange(value)}
+                  status={errors[GeneralFormFieldId.SourceProvider] && MenuToggleStatus.danger}
+                />
+              ),
+            }}
+          />
+
+          <ControlledFormGroup
+            isRequired
+            fieldId={GeneralFormFieldId.TargetProvider}
+            label={generalFormFieldLabels[GeneralFormFieldId.TargetProvider]}
+            controller={{
+              rules: { required: t('Target provider is required.') },
+              render: ({ field }) => (
+                <ProviderSelect
+                  placeholder={t('Select target provider')}
+                  id={GeneralFormFieldId.TargetProvider}
+                  namespace={providerNamespace}
+                  value={(field.value as V1beta1Provider)?.metadata?.name || ''}
+                  onSelect={(_, value) => {
+                    field.onChange(value);
+
+                    if (targetProject) {
+                      setValue(GeneralFormFieldId.TargetProject, '', { shouldValidate: true });
+                    }
+                  }}
+                  status={errors[GeneralFormFieldId.TargetProvider] && MenuToggleStatus.danger}
+                />
+              ),
+            }}
+          />
+
+          <ControlledFormGroup
+            isRequired
+            fieldId={GeneralFormFieldId.TargetProject}
+            label={generalFormFieldLabels[GeneralFormFieldId.TargetProject]}
+            controller={{
+              rules: { required: t('Target project is required.') },
+              render: ({ field }) => (
+                <div ref={field.ref}>
+                  <TypeaheadSelect
+                    isScrollable
+                    placeholder={
+                      targetProviderProjectsLoading ? t('Loading...') : t('Select target project')
+                    }
+                    id={GeneralFormFieldId.TargetProject}
+                    selectOptions={targetProviderProjects.map((project) => ({
+                      value: project.name,
+                      content: project.name,
+                    }))}
+                    isDisabled={targetProviderProjectsLoading}
+                    selected={field.value}
+                    onSelect={(_, value) => field.onChange(value)}
+                    onClearSelection={() => field.onChange('')}
+                    {...(!targetProvider && {
+                      noOptionsAvailableMessage: t(
+                        'Select a target provider to list available target projects',
+                      ),
+                    })}
+                    toggleProps={{
+                      status: errors[GeneralFormFieldId.TargetProject] && MenuToggleStatus.danger,
+                    }}
+                  />
+                </div>
+              ),
+            }}
+          />
+        </FormSection>
+      </Form>
+    </WizardStepContainer>
+  );
+};
+
+export default GeneralInformationStep;
```

Comment:
Please split this file; it's getting too long

Reply:
I've split out the fields that required a lot of the extra constants at the top of the component, and reduced the size to less than 100 lines.